### ISSUE TITLE
moment 2.19.0: "/min/moment-with-locales.min.js" has require() statement incompatible with React Native 0.49

### ISSUE THREAD

**Issue by jspiers** [moment/moment#4235]
I'm trying to build a fork of "react-native-gifted-chat", which depends on moment, using RN 0.49.3.
I am getting the following error:
"Error: <foo>/node_modules/moment/min/moment-with-locales.min.js: require() must have a single string literal argument"

I know a fix/workaround for this same error was included for the file "/moment.js" in moment 2.19.0, but this same issue also affects "/min/moment-with-locales.min.js"

Thanks

**Comment by yurykorzun** [comment]
This issue prevents from upgrading to React Native 0.49

**Comment by ppatel890** [comment]
Having this issue too. Have updated moment to 2.19.1. All other dependencies also using 2.19.1. Anyone have any luck with a fix?

**Comment by maggiepint** [comment]
does this work with 2.19.1? That require statement was changed in that build.

**Comment by ppatel890** [comment]
Yep, still getting it with 2.19.1. Included a picture of npm list moment:

![screen shot 2017-10-11 at 2 48 43 pm](https://user-images.githubusercontent.com/7905655/31468622-629e2446-ae93-11e7-8c26-acaa9e2fe6f9.png)

**Comment by hbk671104** [comment]
Upgraded to 2.19.1 and got the same error, please help. I
![wx20171012-111937](https://user-images.githubusercontent.com/5074788/31477696-556b56c4-aed2-11e7-9bca-244294742675.png)

**Comment by samuelbriole** [comment]
#4187

**Comment by jspiers** [comment]
@samuelbriole thanks yes it seems this issue was already raised in #4187 

Disclaimer: I'm an RN noob.

My understanding is that the problem is that react-native-gifted-chat imports the "min" version of the moment library instead of the top-level ```moment.js```, and while ```moment.js``` has been fixed with respect to the RN 0.49 require(string literal) issue, the "min" version has not been patched because it is supposed to be compiled from the top-level ```moment.js``` which has.

I've tried locally editing ```react-native-gifted-chat/src/GiftedChat.js```:
```import moment from 'moment/min/moment-with-locales.min';```
to
```import moment from 'moment/moment';```

When I do that, I am still getting a "Error: require() must have a single string literal argument" but it happens during bundling and appears to be unrelated to ```moment```.

I'm a RN noob I think it has to do with ```metro-bundler``` issues discussed elsewhere.

TL;DR I'm going to close this issue as a duplicate of #4187

**Comment by jspiers** [comment]
FYI this fix worked for me:
https://github.com/FaridSafi/react-native-gifted-chat/pull/605

**Comment by zachrnolan** [comment]
@jspiers It looks like you're having the exact same issue as me (using gifted chat as well). Would you mind summarizing exactly what you did to get this all working? (i.e. what version of each library you are currently on: react-native-gifted-chat, react-native, moment).
